### DISCUSSION TITLE
Replace image colors to another

### DISCUSSION THREAD

**Discussion by ivanamar996** [dlemstra/Magick.NET#1325] [Help]
Hi,

I've tried to replace all colors from image to another by using Opaque method, this is code snippet of it:

![image](https://user-images.githubusercontent.com/39298540/209931945-1cfd5bb2-41e3-48f0-88dc-67f3bac98885.png)

Target color is : 
![image](https://user-images.githubusercontent.com/39298540/209932337-5b91ce1c-3ce6-4369-b933-aaad28e49e49.png)

For source PNG image that I uploaded, the result is not correct since all colors are replaced to gray. And I am confused why this is happening. I also tried to call imagemagick from command line and colors are replaced to the correct red color.

SOURCE IMAGE:
![473b9016-4b17-41e9-bdb0-e0331f546abf-png](https://user-images.githubusercontent.com/39298540/209933375-87591572-14cb-4a1f-a884-5886e601778e.png)

RESULT:
![result](https://user-images.githubusercontent.com/39298540/209932963-b4ec0097-8ddf-4a9b-9b80-be893f0fc22a.png)

I tried to run image magick command in cmd:
magick source.png -fuzz 30% -fill rgba(214,0,88,85) -opaque black TECH2.png

And result is:
![TECH2](https://user-images.githubusercontent.com/39298540/209934067-8c254b03-a119-43f1-aac5-123d574e7de5.png)

Is there anything that I am doing wrong?

**Comment by dlemstra** [comment]
It appears that your source image is grayscale. What happens when you change the `ColorType` of the `image` to `ColorType.TrueColorAlpha` at the start?

**Comment by ivanamar996** [comment]
Thanks a lot, setting ColorType to TrueColorAlpha fixed my issue.
